ose/environment.py

```diff
@@ -20,12 +20,25 @@ def __init__(self, students, statements=None):
             A list of statements in xAPI format
         """
         self.students = dict()
-        self._create_students_hash(students)
-        if statements is not None :
-            self.statements = _create_students_statements(statements)
-            self._update_students_statements()
-        else :
-            self.statements = {s.name : [] for s in self.students.values()}
+        self.statements = defaultdict(list)
+
+        for s in students:
+            self.add_student(s)
+        if statements is not None:
```

Comment:
Better to define `[]` as the default for statements and remove this line. Could also be the defulat for students.